lxd/storage_utils.go

```diff
@@ -181,13 +181,25 @@ func lxdUsesPool(dbObj *sql.DB, onDiskPoolName string, driver string, onDiskProp
 	return false, "", nil
 }
 
+func xfsGenerateNewUUID(path string) error {
+	_, err := shared.RunCommand("xfs_admin", "-U", "generate", path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func makeFSType(path string, fsType string) (string, error) {
```

Comment:
This function is never called from within the LVM codebase, so I'm unsure how you expect LVM to have its UUID re-generated when needed.

Reply:
Uhm, it must be otherwise no volume would have a file system and Jenkins would look very very very sad. I'm sure it's called in the appropriate helpers in one of the storage_lvm_$ files.